### ISSUE TITLE
Vienna 3.8 beta – Applescript: cannot get HTML source

### ISSUE THREAD

**Issue by phiw13** [ViennaRSS/vienna-rss#1523]
A basic applescript using `get documentHTMLSource` to open the HTML source of an article in a code editor fails (per the Vienna AS dictionary).

Result: the script opens a blank document. This fails independently of the “new“ browser (WKWebView). It worked fine in 3.7.x.

The following script opens the HTML source in a new BBEdit document (I have a similar one for SubEthaEdit).

```
tell application "Vienna"
	set this_source to get documentHTMLSource
	tell application "BBEdit"
		activate
		make new text window with properties {text:this_source, source language:"HTML"}
		select insertion point before text of front text document
	end tell
end tell
```

PS – this is a bit annoying for debugging feeds as the Web inspector cannot be used in WKWebView, unless I miss a magic incantation. The Web inspector still works in the “old” browser.

**Comment by phiw13** [comment]
> I am not sure if the issue is due to a mistake in the Article View or due to the WebView behaving differently when a local document is loaded. It is not intentional.

hmm. Hopefully there is some sort of workaround?

> For the Web Inspector,[…]Theoretically that could be made into a setting in Viennas Preferences.

Now that would be quite nice, I suspect I am not the only one who uses the inspector tool.

**Comment by Eitot** [comment]
The web inspector within Vienna has never worked for me; it just overlays an opaque view onto the webview. I am not keen on exposing this to release builds.

<img width="724" alt="Screenshot 2021-10-30 at 13 17 12" src="https://user-images.githubusercontent.com/15073177/139530818-aa5ed73d-b740-428a-b78a-bd9d7d39ee5e.png">
<img width="719" alt="Screenshot 2021-10-30 at 13 17 26" src="https://user-images.githubusercontent.com/15073177/139530827-a513ff01-589e-4398-9205-25c1c7d0fa7e.png">

**Comment by phiw13** [comment]
Didn’t this command work for you ? (quit Vienna beforehand…)

`defaults write uk.co.opencommunity.vienna2 WebKitDeveloperExtras -bool true`

That has worked for a long time, and still does as long as you have the “old” browser enabled. Right click any element in article view, you’ll see the “Inspect element” entry in the context menu.

<img width="904" alt="Vienna Inspector" src="https://user-images.githubusercontent.com/8944809/139531526-c4025efa-65a4-4065-b2c9-4abf892525a7.png">

**Comment by Eitot** [comment]
I have only tested it in developer builds with the WebKit (“new”) browser enabled. The `WebKitDeveloperExtras` default makes no difference.

**Comment by phiw13** [comment]
Yes with WKWebView that command does not work (as alluded in my OP), lots of discussions on stackoverflow about that. That is why I turned back to my Old Applescript(s) only that does not work either (In Vienna; I have a AppleScript working with NNW).

**Comment by Eitot** [comment]
I was merely pointing out that exposing the web inspector private API via a toggle in advanced preferences, as suggested by TAKeanice, is not an option. Instead, the AppleScript function `documentHTMLSource` has to be fixed, so that it also returns the page source of the article, as it does in Vienna 3.7 and older.

**Comment by barijaona** [comment]
For me, access to debugging is possible through **Safari**'s Development menu.
Once this menu is activated (cf. Safari's advanced preferences), a submenu with your machine's name appears into it : Vienna's tab are listed there and can be selected.

You probably also need the `defaults write uk.co.opencommunity.vienna2 WebKitDeveloperExtras -bool true` command beforehand.

**Comment by phiw13** [comment]
this is still relevant, FWIW

**Comment by barijaona** [comment]
This will be fixed in 3.9.0, at least for recent versions macOS Ventura (commit [e0f288b2ed57ec5e486300a2b3555bc57fe2c000](https://github.com/ViennaRSS/vienna-rss/commit/e0f288b2ed57ec5e486300a2b3555bc57fe2c000#diff-be10486530e818fb183f8a0804291000b3b7670be277a0bc106814d3c17a9a9bR102-R104) included in PR #1666)
